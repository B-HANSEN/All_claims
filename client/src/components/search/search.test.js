import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Search from '.';

describe('Search Page tests', () => {
	it('should match snapshot', () => {
		const renderer = new ShallowRenderer();
		const result = renderer.render(<Search setValue={jest.fn()} />);
		expect(result).toMatchSnapshot();
	});

	it('should change input value when user types and set local state', () => {
		const handleValue = jest.fn();
		render(<Search setValue={handleValue} />);

		const input = screen.getByTestId('search');
		expect(input).toBeInTheDocument();

		fireEvent.change(input, { target: { value: 'CL-49085' } });
		expect(input.value).toBe('CL-49085');

		const submit = screen.getByRole('button', {
			name: /submit/i,
		});
		expect(submit).toBeEnabled();
		fireEvent.click(submit);

		expect(handleValue).toHaveBeenCalledTimes(1);
	});
});
