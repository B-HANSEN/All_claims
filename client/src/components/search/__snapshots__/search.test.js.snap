// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page tests should match snapshot 1`] = `
<Form
  onSubmit={[Function]}
>
  <FormGroup
    className="mb-3"
  >
    <FormLabel>
      Search by ID, holder name or policy
    </FormLabel>
    <FormControl
      data-testid="search"
      id="search"
      name="search"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Enter query here..."
      size="lg"
      type="text"
      value=""
    />
  </FormGroup>
  <Button
    className="mb-2"
    onClick={[Function]}
    size="lg"
    type="submit"
    variant="primary"
  >
    Submit
  </Button>
   
  <Button
    className="mb-2"
    onClick={[Function]}
    size="lg"
    type="submit"
    variant="outline-secondary"
  >
    Reset
  </Button>
</Form>
`;
